mlflow/store/model_registry/file_store.py

```diff
@@ -0,0 +1,767 @@
+    ):
+        """
+        Search for registered models in backend that satisfy the filter criteria.
+
+        :param filter_string: Filter query string, defaults to searching all registered models.
+        :param max_results: Maximum number of registered models desired.
+        :param order_by: List of column names with ASC|DESC annotation, to be used for ordering
+                         matching search results.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                            a ``search_registered_models`` call.
+        :return: A PagedList of :py:class:`mlflow.entities.model_registry.RegisteredModel` objects
+                that satisfy the search expressions. The pagination token for the next page can be
+                obtained via the ``token`` attribute of the object.
+        """
+        if not isinstance(max_results, int) or max_results < 1:
+            raise MlflowException(
+                "Invalid value for max_results. It must be a positive integer,"
+                f" but got {max_results}",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        if max_results > SEARCH_REGISTERED_MODEL_MAX_RESULTS_THRESHOLD:
+            raise MlflowException(
+                "Invalid value for request parameter max_results. "
+                "It must be at most {}, but got value {}".format(
+                    SEARCH_REGISTERED_MODEL_MAX_RESULTS_THRESHOLD, max_results
+                ),
+                INVALID_PARAMETER_VALUE,
+            )
+
+        registered_models = self._list_all_registered_models()
+        filtered_rms = SearchModelUtils.filter(registered_models, filter_string)
+        sorted_rms = SearchModelUtils.sort(filtered_rms, order_by)
+        # we query for max_results + 1 items to check whether there is another page to return.
+        # this remediates having to make another query which returns no items.
+        max_results_for_query = max_results + 1
+        offset = SearchUtils.parse_start_offset_from_page_token(page_token)
+
+        if page_token:
+            sorted_rms = sorted_rms[offset:]
+
+        def compute_next_token(current_size):
+            next_token = None
+            if max_results_for_query == current_size:
+                final_offset = offset + max_results
+                next_token = SearchUtils.create_page_token(final_offset)
+            return next_token
+
+        next_page_token = compute_next_token(len(sorted_rms))
+        return PagedList(sorted_rms[:max_results], next_page_token)
+
+    def get_registered_model(self, name):
+        """
+        Get registered model instance by name.
+
+        :param name: Registered model name.
+        :return: A single :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        try:
+            return self._get_registered_model(name)
+        except MlflowException as e:
+            if e.error_code == RESOURCE_DOES_NOT_EXIST:
```

Comment:
This check doesn't quite work as-is because `e.error_code` is a string while `RESOURCE_DOES_NOT_EXIST` is an `int`.

I think we need 

```suggestion
            if e.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST):
```